util/download_util.py

```diff
@@ -166,14 +173,16 @@ def request_url(url: str,
                 # Exception in case of API error.
                 return None
             except (requests.exceptions.ConnectTimeout,
-                    urllib.error.URLError) as e:
-                # Exception when server is overloaded, retry after a delay
-                if attempt >= retries:
-                    raise urllib.error.URLError
-                else:
-                    logging.debug(
-                        f'Retrying URL {url} after {retry_secs} secs ...')
-                    time.sleep(retry_secs)
+                    requests.exceptions.ConnectionError, urllib.error.URLError,
+                    urllib.error.HTTPError) as e:
+                logging.debug(f'Got exception {e} for {url}, {params}')
+
+            # retry in case of errors
+            if attempt >= retries:
+                raise urllib.error.URLError
```

Comment:
Should we not just bubble up the original error instead using a fixed error. This would help debug without enabling debug logs.

Reply:
We would like to retry on errors like timeout, server error etc which are also exceptions hence the original error is consumed with a log message and retry attempted.